contracts/src/CrossChainApplications/examples/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -230,8 +230,12 @@ contract NativeTokenDestination is TeleporterOwnerUpgradeable, INativeTokenDesti
             );
         }
 
-        uint256 burnedDifference = BURNED_TX_FEES_ADDRESS.balance - latestBurnedFeesReported;
+        uint256 burnedAddressBalance = BURNED_TX_FEES_ADDRESS.balance;
+        uint256 burnedDifference = burnedAddressBalance - latestBurnedFeesReported;
```

Comment:
I think we should `require` that `burnedAddressBalance >= latestBurnedFeesReported`, or use safemath. If the destination somehow becomes corrupted such that `BURNED_TX_FEES_ADDRESS.balance` decreases, that would cause an underflow and, and up to INTMAX reported burned fees to the source chain.

Reply:
Yeah, I don't think this should be possible under any circumstance, but having the sanity check is worth it based on the catastrophic consequences.

Reply:
Thinking about it some more, a scenario in which the destination chain is corrupted such that the burn address is compromised, but arbitrary Teleporter messages are still not able to be sent seems not that likely. Still a good idea to prevent an invalid value here. Thanks for the fix.